### PR TITLE
fix: version defaulting to `0.0.0.dev0`

### LINKED ISSUE #37: [BUG] check update seems to be wrong
**The `slack_cli_hooks` versions**

**Python runtime version**

**OS info**

**Describe the bug**

If the installed package is the latest version, it seem like the check_update script reports this as there being a newer version

**Steps to reproduce**

1. pip install slack-cli-hooks
2. got to a valid project
3. python3 -m slack_cli_hooks.hooks.check_update | jq .

**Expected result**

```json
{
  "name": "Slack Bolt",
  "url": "https://api.slack.com/automation/changelog",
  "releases": [
    {
      "name": "slack_cli_hooks",
      "current": "0.0.0",
      "latest": "0.0.0",
      "update": false,
      "breaking": false
    },
    {
      "name": "slack_bolt",
      "current": "1.19.0",
      "latest": "1.18.1",
      "update": false,
      "breaking": false
    },
    {
      "name": "slack_sdk",
      "current": "3.27.1",
      "latest": "3.27.1",
      "update": false,
      "breaking": false
    }
  ]
}
```

**Actual result**
```json
{
  "name": "Slack Bolt",
  "url": "https://api.slack.com/automation/changelog",
  "releases": [
    {
      "name": "slack_cli_hooks",
      "current": "0.0.0",
      "latest": "0.0.0",
      "update": true,
      "breaking": false
    },
    {
      "name": "slack_bolt",
      "current": "1.19.0",
      "latest": "1.18.1",
      "update": false,
      "breaking": false
    },
    {
      "name": "slack_sdk",
      "current": "3.27.1",
      "latest": "3.27.1",
      "update": false,
      "breaking": false
    }
  ]
}
```
```zsh
└─[$] <git:(main*)> hermes upgrade
   
🛠️  An update from Slack Bolt is available:

    › slack_cli_hooks
      0.0.0 → 0.0.0

   For more information about this update, visit:
   https://api.slack.com/automation/changelog
```

**Requirements**

Please read the [Contributing guidelines](https://github.com/slackapi/python-slack-hooks/blob/main/.github/CONTRIBUTING.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to those rules.

### PR DESCRIPTION
**Pull request by WilliamBergamin**
### Summary

This PR aims to fix #37, it appear the value being used to look up the version of the package on the users machine was defaulting to `0.0.0.dev0` making the `check_update` command unusable. These changes revert some of the automation introduced in #36 to fix this bug.

### Testing

1. pull the changes locally
2. build the package
3. test the package on an app with `slack create -t https://github.com/slack-samples/bolt-python-custom-function-template`

### Requirements 

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/python-slack-hooks/blob/main/.github/CONTRIBUTING.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've run `./scripts/install_and_run_tests.sh` after making the changes.

### CODE REVIEW

File: .github/maintainers_guide.md [hunk 1]
```diff
@@ -108,33 +108,47 @@ If you want to test the package locally you can.
...
+     - `.dev` will indicate to pip that this is a
+       [Development Release](https://peps.python.org/pep-0440/#developmental-releases)
+     - Note that the `dev` version can be bumped in development releases:
+       `1.2.3.dev0` -> `1.2.3.dev1`
+   - Commit with a message including the new version number. For example
+     `1.2.3.dev0` & Push the commit to a branch where the development release
+     will live (create it if it does not exist)
+     - `git checkout -b future-release`
+     - `git commit -m 'version 1.2.3.dev0'`
+     - `git push future-release`
```
Comment (zimeg) [inline .github/maintainers_guide.md:145]: Always fascinating to see the ways to push upstream haha! No change needed, I'm just a big fan of

```
$ git push -u origin future-release
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -22,8 +22,6 @@ jobs:
         run: |
           python -m build
           twine check dist/*
-        env:
-          SLACK_CLI_HOOKS_VERSION: ${{ github.event.release.name }}
```
Comment (zimeg) [inline .github/workflows/release.yml:26]: 🫖 Pouring one out

File: slack_cli_hooks/hooks/check_update.py [hunk 3]
```diff
@@ -30,8 +30,8 @@ def __init__(
     ):
         self.name = name
         if current and latest:
-            self.current = current.base_version
-            self.latest = latest.base_version
+            self.current = str(current)
+            self.latest = str(latest)
             self.update = current < latest
             self.breaking = (current.major - latest.major) != 0
```
Comment (zimeg) [inline slack_cli_hooks/hooks/check_update.py:36]: Could a test also be added for this? We might want to also avoid major downgrades(??) with

```
latest.major - current.major > 0
```

**Review by zimeg — APPROVED** [review]
Oh fascinating catch but a bummer that the automation magic goes away for now... Totally missed that this would fallback to the dev version by default!

File: slack_cli_hooks/hooks/check_update.py (hunk 3 above)
Comment (seratch) [reply slack_cli_hooks/hooks/check_update.py:36]: Yeah, I was also wondering about the breaking property's behavior. Adding more various patterns to the test is safer. Other than that, this PR looks good to me

File: .github/maintainers_guide.md [hunk 4]
```diff
@@ -108,33 +108,47 @@ If you want to test the package locally you can.
...
 ```
 
 #### Development Deployment
 
-Releases for this library are automatically generated off of git releases. Before
-creating a new release, ensure that everything on a stable branch has landed, then
-[run the tests](#run-all-the-unit-tests).
-
-1. Create a new GitHub Release from the
+Releases for this library are automatically generated off of git releases.
```
Comment (filmaj) [inline .github/maintainers_guide.md:125]: `git` has no concept of releases, only tags. Did you mean GitHub Releases here?

File: .github/maintainers_guide.md [hunk 5]
```diff
@@ -108,33 +108,47 @@ If you want to test the package locally you can.
...
+     will live (create it if it does not exist)
+     - `git checkout -b future-release`
+     - `git commit -m 'version 1.2.3.dev0'`
+     - `git push future-release`
+2. Create a new GitHub Release from the
    [Releases page](https://github.com/slackapi/python-slack-hooks/releases) by
    clicking the "Draft a new release" button.
-2. Input a new version manually into the "Choose a tag" input. You can start off
-   by incrementing the version to reflect a patch. (i.e. 1.16.0 -> 1.16.1.dev0)
+3. Input a the version manually into the "Choose a tag" input. You must use the
```
Comment (filmaj) [inline .github/maintainers_guide.md:149]:
```suggestion
3. Input the version manually into the "Choose a tag" input. You must use the
```

File: .github/maintainers_guide.md [hunk 6]
```diff
@@ -143,74 +157,70 @@ creating a new release, ensure that everything on a stable branch has landed, th
...
    <https://pypi.org/project/slack-cli-hooks/>.
-9. (Slack Internal) Communicate the release internally
+10. (Slack Internal) Communicate the release internally
 
 #### Production Deployment
 
-Releases for this library are automatically generated off of git releases. Before
-creating a new release, ensure that everything on the `main` branch since the
-last tag is in a releasable state! At a minimum,
+Releases for this library are automatically generated off of git releases.
```
Comment (filmaj) [inline .github/maintainers_guide.md:179]: Same thing here, `git tag`s or GitHub Releases?

**Review by filmaj — CHANGES_REQUESTED** [review]
Some minor docs comments left but agreed with Ethan's point re: major downgrades, good test to add.

File: slack_cli_hooks/hooks/check_update.py (hunk 3 above)
Comment (WilliamBergamin) [reply slack_cli_hooks/hooks/check_update.py:36]: Good catch 💯 not sure how this got in, fixed and added more tests around it

File: .github/maintainers_guide.md (hunk 4 above)
Comment (WilliamBergamin) [reply .github/maintainers_guide.md:125]: @filmaj good catch 💯  currently the [Github Action that deploys a new version](https://github.com/slackapi/python-slack-hooks/blob/main/.github/workflows/release.yml) of the project to Pypi is triggered by publishing a new Github Release, the instructions ask to create a tag along with the GitHub release
Should this action be triggered by the tag creation instead?

Comment (filmaj) [reply .github/maintainers_guide.md]: I think Release creation is a nice underlying trigger, as it forces the maintainer to create a GitHub Release (which in turn will create a tag).

**Review by filmaj — APPROVED** [review]

**Pull request merged**